app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -2078,25 +2092,36 @@ private void showClearingQueueConfirmation(final Runnable onAllow) {
     }
 
     private void showExternalPlaybackDialog() {
-        if (sortedVideoStreams == null) {
+        if (currentInfo == null) {
             return;
         }
-        final CharSequence[] resolutions = new CharSequence[sortedVideoStreams.size()];
-        for (int i = 0; i < sortedVideoStreams.size(); i++) {
-            resolutions[i] = sortedVideoStreams.get(i).getResolution();
-        }
-        final AlertDialog.Builder builder = new AlertDialog.Builder(activity)
-                .setNegativeButton(R.string.cancel, null)
-                .setNeutralButton(R.string.open_in_browser, (dialog, i) ->
-                        ShareUtils.openUrlInBrowser(requireActivity(), url)
-                );
-        // Maybe there are no video streams available, show just `open in browser` button
-        if (resolutions.length > 0) {
-            builder.setSingleChoiceItems(resolutions, selectedVideoStreamIndex, (dialog, i) -> {
+        final List<VideoStream> videoStreams = removeNonUrlAndTorrentStreams(
+                currentInfo.getVideoStreams());
+        final List<VideoStream> videoOnlyStreams = removeNonUrlAndTorrentStreams(
+                currentInfo.getVideoOnlyStreams());
+        final List<VideoStream> sortedVideoStreams = ListHelper.getSortedStreamVideosList(
+                activity, videoStreams, videoOnlyStreams, false);
+        final AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+        builder.setTitle(R.string.select_quality_external_player);
+        builder.setNegativeButton(android.R.string.cancel, null);
+        builder.setNeutralButton(R.string.open_in_browser, (dialog, i) ->
+                ShareUtils.openUrlInBrowser(requireActivity(), url));
+        if (!sortedVideoStreams.isEmpty()) {
```

Comment:
Inverting this statement makes it easier to read

Reply:
Done